lib/ClangImporter/ImportDecl.cpp

```diff
@@ -8682,10 +8682,20 @@ void ClangImporter::Implementation::importAttributes(
     if (method->isDirectMethod() && !AnyUnavailable) {
       assert(isa<AbstractFunctionDecl>(MappedDecl) &&
              "objc_direct declarations are expected to be an AbstractFunctionDecl");
-      MappedDecl->getAttrs().add(new (C) FinalAttr(/*IsImplicit=*/true));
-      if (auto accessorDecl = dyn_cast<AccessorDecl>(MappedDecl)) {
-        auto attr = new (C) FinalAttr(/*isImplicit=*/true);
-        accessorDecl->getStorage()->getAttrs().add(attr);
+      if (isa<ConstructorDecl>(MappedDecl)) {
+        // TODO: Teach Swift how to directly call these functions.
+        auto attr = AvailableAttr::createPlatformAgnostic(
```

Comment:
I'm a little surprised we haven't extracted a helper for this, but we have not, in fact, extracted a helper for this. (`markUnavailable()` uses plain `PlatformAgnosticAvailabilityKind::Unavailable`—although perhaps it shouldn't? Probably a question for another day.)

Nothing you need to do here, just thinking about how we might improve things.